kubeprod/pkg/eks/platform.go

```diff
@@ -50,20 +49,18 @@ func (conf *Config) getAwsSession() *session.Session {
 				session.Options{
 					// Load AWS SDK configuration parameters (including the AWS region)
 					SharedConfigState: session.SharedConfigEnable,
-					Config: *aws.NewConfig().WithHTTPClient(&http.Client{
-						// Configure an explicit time-out of 30 seconds
-						Timeout: 30 * time.Second,
-					}),
+					// Prompt on stdin for MFA token if required
+					AssumeRoleTokenProvider: stscreds.StdinTokenProvider,
 				}))
 	}
 	return conf.session
 }
 
 // Retrieves the identity of the caller. Among other details retrieves
 // the AWS account number.
-func (conf *Config) getCallerIdentity() (*sts.GetCallerIdentityOutput, error) {
+func (conf *Config) getCallerIdentity(ctx context.Context) (*sts.GetCallerIdentityOutput, error) {
```

Comment:
Why don't you put the `ctx` variable inside the `conf` structure? This way you save passing it around in functions like `getUserPolicy()`, right?